src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,49 @@
 
 class Animal {
   // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (this.hidden === true) {
+      this.hidden = false;
+    };
+    this.hidden = true;
+  }
```

Comment:
```suggestion
    this.hidden = true;
  }
```

this method should not toggle hidden value and only change it to true